Source/gendung.cpp

```diff
@@ -604,27 +607,25 @@ void DRLG_PlaceThemeRooms(int minSize, int maxSize, int floor, int freq, bool rn
 					if (themeH < min || themeH > max)
 						themeH = min;
 				}
-				themeLoc[themeCount].x = i + 1;
-				themeLoc[themeCount].y = j + 1;
-				themeLoc[themeCount].width = themeW;
-				themeLoc[themeCount].height = themeH;
-				if (IsAnyOf(leveltype, DTYPE_CAVES, DTYPE_NEST))
-					DRLG_RectTrans(2 * i + 20, 2 * j + 20, 2 * (i + themeW) + 15, 2 * (j + themeH) + 15);
-				else
-					DRLG_MRectTrans(i + 1, j + 1, i + themeW, j + themeH);
+				themeLoc[themeCount].room = { { i + 1, j + 1 }, { themeW, themeH } };
+				if (IsAnyOf(leveltype, DTYPE_CAVES, DTYPE_NEST)) {
+					DRLG_RectTrans({ Point(i, j).megaToWorld() + Displacement { 4, 4 }, { themeW * 2 - 5, themeH * 2 - 5 } });

```

Comment:
Fixing the loop bounds would let you make this symmetric
```suggestion
					DRLG_RectTrans({ Point(i, j).megaToWorld() + Displacement { 4, 4 }, { themeW * 2 - 4, themeH * 2 - 4 } });
```
However this is effectively doing the Mega -> World transformation manually, so you could also do:
```suggestion
					DRLG_MRectTrans(Point { i, j } + Displacement { 2, 2 }, { themeW - 2, themeH - 2 } });
```

Reply:
Thanks that made this a lot cleaner :)

Reply:
Wait that second one actually fails, since it's 1x1 shorter then the original :) I'll apply the first one and we can try to align things in a later PR :pray:

Reply:
Yeah that still requires fixing the off-by-one in RectTrans (and removing the size - 1 in MRectRans) :D

Reply:
I have a strong feeling that there are tons of bugs in how the sections are assigned. It's something we can look in to fixing but preferably where that is the main focus since it will require updating the figures and will need some additional testing since it can affect other parts of the game.